cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -247,3 +255,46 @@ func (m *DDLSink) Close() {
 		}
 	}
 }
+
+// asyncExecAddIndexDDLIfTimeout executes ddl in async mode.
+func (m *DDLSink) asyncExecAddIndexDDLIfTimeout(ctx context.Context, ddl *model.DDLEvent) error {
+	done := make(chan error, 1)
+	// wait for 1 second at most
+	tick := time.NewTimer(1 * time.Second)
+	defer tick.Stop()
+	log.Info("async exec ddl start",
+		zap.String("changefeedID", m.id.String()),
+		zap.Uint64("commitTs", ddl.CommitTs),
+		zap.String("ddl", ddl.Query))
+	go func() {
+		if err := m.execDDLWithMaxRetries(ctx, ddl); err != nil {
+			log.Error("async exec ddl failed",
+				zap.String("changefeedID", m.id.String()),
+				zap.Uint64("commitTs", ddl.CommitTs),
+				zap.String("ddl", ddl.Query))
+			done <- err
+			return
+		}
+		log.Info("async exec ddl done",
+			zap.String("changefeedID", m.id.String()),
+			zap.Uint64("commitTs", ddl.CommitTs))
+		done <- nil
+	}()
+	for {
```

Comment:
The DDL is executing in another goroutine, and this `for` loop looks useless, since the ticker will be triggered 1s later, so the method return, and the error channel `done` is not correctly checked.

Reply:
fixed